units/event_metajob.go

```diff
@@ -239,6 +239,17 @@ func (j *eventMetaJob) dispatch(ctx context.Context, notifications []notificatio
 	return catcher.Resolve()
 }
 
+// dispatchUnprocessedNotifications gets unprocessed notifications
+// leftover by previous runs and dispatches them
+func (j *eventMetaJob) dispatchUnprocessedNotifications(ctx context.Context) error {
+	unprocessedNotifications, err := notification.FindUnprocessed()
+	if err != nil {
+		return errors.Wrap(err, "can't find unprocessed notifications")
+	}
+
+	return j.dispatch(ctx, unprocessedNotifications)
```

Comment:
This approach seems a bit strange to me, since it's possible (if there's a bug) that whatever caused the notification to not send the first time could happen again if we queue it up. Have you considered having event_send also send any unprocessed notifications?

Reply:
Putting the cleanup in event_send was considered, but since sending each notification could potentially take some time (dealing with Jira's API for instance) each event_send job should probably only process a single notification.

The bug that was observed was that the notification was inserted into the database but the job to process it wasn't because the context was canceled. This particular bug would be addressed by having the next event_metajob create new jobs for dropped notifications.